Use Next.JS Site-Preview mode (#1237)
vivid-planet/comet: .changeset/yellow-seahorses-lick.md

---
"@comet/cms-admin": major
"@comet/cms-api": major
"@comet/eslint-config": minor
"@comet/cms-site": minor
---

> Migrate Site-Preview to Next.js Preview Mode
> 
> Requires following changes to site:
> 
> -   Import useRouter from next/router (not exported from @comet/cms-site anymore)
> -   Import Link from next/link (there is no export from @comet/cms-site anymore)
> -   Remove Preview Pages (Pages under preview/ subdirectory which call createGetUniversalProps with preview parameters)
> -   Remove createGetUniversalProps
>     -   Just implement getStaticProps/getServerSideProps (Preview Mode will SSR automatically)
>     -   Get previewData from context and use it to configure the GraphQL-Client
> -   Add SitePreviewProvider to App
> -   Add /api/preview
> 
> Requires following changes to api:
> 
> -   Set sitePreviewSecret in PageTreeModule-options (make sure it's the same for the across multiple api-instances)

Reviewer: ```suggestion
Migrate site preview to Next.js Preview Mode

Requires following changes to site:

-   Import `useRouter` from `next/router` (not exported from `@comet/cms-site` anymore)
-   Import `Link` from `next/link` (not exported from `@comet/cms-site` anymore)
-   Remove preview pages (pages in `src/pages/preview/` directory which call `createGetUniversalProps` with preview parameters)
-   Remove `createGetUniversalProps`
    -   Just implement `getStaticProps`/`getServerSideProps` (Preview Mode will SSR automatically)
    -   Get `previewData` from `context` and use it to configure the GraphQL Client
-   Add `SitePreviewProvider` to `App` (typically in `src/pages/_app.tsx`)
-   Add `/api/preview` Next API route

Requires following changes to API:

-   Set `sitePreviewSecret` in `PageTreeModule`-options (make sure it's the same across multiple API-instances)
```

Author: 4d950a854ce7ea06170e416bab6a930fb776b157
